mws/sitesmanagement/cronjobs.py

```diff
@@ -391,10 +393,38 @@ def send_reminder_delete_upgraded():
                      "old server has not been deleted yet. As the number of test servers the "
                      "MWS can support is limited, this may be blocking others from upgrading.\n"
                      "If you're happy with the new server, please delete the old one by "
-                     "visiting\n\nhttps://panel.mws3.csx.cam.ac.uk%s\n\nand clicking the "
-                     "'Delete the test server' button.\n\nThanks,\nMWS Support\n" % (conf_url),
+                     "visiting\n\n%s%s\n\nand clicking the "
+                     "'Delete the test server' button.\n\nThanks,\nMWS Support\n" % (settings.MAIN_DOMAIN, conf_url),
                 from_email="Managed Web Service Support <%s>"
                            % getattr(settings, 'EMAIL_MWS3_SUPPORT', 'mws-support@uis.cam.ac.uk'),
                 to=[site.email],
                 headers={'Return-Path': getattr(settings, 'EMAIL_MWS3_SUPPORT', 'mws-support@uis.cam.ac.uk')}
             ).send()
+
+
+@shared_task(base=ScheduledTaskWithFailure)
+def dequeue_upgrades():
+    '''
+    Create a test server for a site popped off the upgrade queue and notify
+    site administrators.
+    '''
+    pending_upgrades = len([x for x in Service.objects.filter(type='test').prefetch_related('virtual_machines') if x.active])
```

Comment:
not sure why you prefetch virtual_machines

Reply:
to speed up the check x.active, which counts virtual machines for the service.